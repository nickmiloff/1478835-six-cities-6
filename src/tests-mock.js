export const simpleMock = `test`;

export const cards = {
  changedCard: {
    id: 2,
    isFavorite: true
  },
  start: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 2,
      isFavorite: false
    },
    {
      id: 3,
      isFavorite: false
    }
  ],
  end: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 2,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: false
    }
  ],
  endWithRemove: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: false
    }
  ]
};

export const auth = {
  api: {
    "id": 1,
    "email": `mail@mail.com`,
    "name": `name`,
    "avatar_url": `avatar.jpeg`,
    "is_pro": true
  },
  adapt: {
    id: 1,
    email: `mail@mail.com`,
    name: `name`,
    avatarUrl: `avatar.jpeg`,
    isPro: true
  }
};

export const user = {
  email: `mail@mail.com`,
  password: 12345678
};

export const cardsFull = {
  api: [
    {
      "city": {
        "name": `Paris`,
        "location": {
          "latitude": 48.85661,
          "longitude": 2.351499,
          "zoom": 13
        }
      },
      "images": [],
      "title": `Tile House`,
      "rating": 2,
      "type": `house`,
      "bedrooms": 2,
      "price": 161,
      "goods": [`Laptop friendly workspace`],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      "location": {
        "latitude": 48.85761,
        "longitude": 2.358499,
        "zoom": 16
      },
      "id": 1,
      "is_favorite": false,
      "is_premium": true,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      "max_adults": 9
    },
    {
      "city": {
        "name": `Amsterdam`,
        "location": {
          "latitude": 52.37454,
          "longitude": 4.897976,
          "zoom": 13
        }
      },
      "images": [],
      "title": `Loft Studio in the Central Area`,
      "rating": 3.6,
      "type": `apartment`,
      "bedrooms": 3,
      "price": 271,
      "goods": [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`
      ],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "location": {
        "latitude": 52.388540000000006,
        "longitude": 4.899976,
        "zoom": 16
      },
      "id": 2,
      "is_favorite": false,
      "is_premium": false,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      "max_adults": 8
    },
    {
      "city": {
        "name": `Paris`,
        "location": {
          "latitude": 48.85661,
          "longitude": 2.351499,
          "zoom": 13
        }
      },
      "images": [],
      "title": `Tile House`,
      "rating": 4.6,
      "type": `house`,
      "bedrooms": 2,
      "price": 865,
      "goods": [`Laptop friendly workspace`, `Breakfast`],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "location": {
        "latitude": 48.83861,
        "longitude": 2.350499,
        "zoom": 16
      },
      "id": 3,
      "is_favorite": true,
      "is_premium": false,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      "max_adults": 8
    }
  ],
  adapted: [
    {
      city: {
        name: `Paris`,
        location: {
          latitude: 48.85661,
          longitude: 2.351499,
          zoom: 13,
        },
      },
      images: [],
      title: `Tile House`,
      rating: 2,
      type: `house`,
      bedrooms: 2,
      price: 161,
      goods: [`Laptop friendly workspace`],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true,
      },
      description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      location: {
        latitude: 48.85761,
        longitude: 2.358499,
        zoom: 16,
      },
      id: 1,
      isFavorite: false,
      isPremium: true,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      maxAdults: 9,
    },
    {
      city: {
        name: `Amsterdam`,
        location: {
          latitude: 52.37454,
          longitude: 4.897976,
          zoom: 13,
        },
      },
      images: [],
      title: `Loft Studio in the Central Area`,
      rating: 3.6,
      type: `apartment`,
      bedrooms: 3,
      price: 271,
      goods: [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`,
      ],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true,
      },
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      location: {
        latitude: 52.388540000000006,
        longitude: 4.899976,
        zoom: 16,
      },
      id: 2,
      isFavorite: false,
      isPremium: false,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      maxAdults: 8,
    },
    {
      city: {
        name: `Paris`,
        location: {
          latitude: 48.85661,
          longitude: 2.351499,
          zoom: 13,
        },
      },
      images: [],
      title: `Tile House`,
      rating: 4.6,
      type: `house`,
      bedrooms: 2,
      price: 865,
      goods: [`Laptop friendly workspace`, `Breakfast`],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true,
      },
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      location: {
        latitude: 48.83861,
        longitude: 2.350499,
        zoom: 16,
      },
      id: 3,
      isFavorite: true,
      isPremium: false,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      maxAdults: 8,
    },
  ]
};

export const comments = {
  api: [
    {
      "comment": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      "date": `2019-05-08T14:13:56.569Z`,
      "id": 1,
      "rating": 4,
      "user": {
        "avatar_url": `img/1.png`,
        "id": 4,
        "is_pro": false,
        "name": `Max`
      }
    },
    {
      "comment": `A quiet of Amsterdam.`,
      "date": `2019-06-08T14:13:56.569Z`,
      "id": 2,
      "rating": 3,
      "user": {
        "avatar_url": `img/2.png`,
        "id": 5,
        "is_pro": false,
        "name": `Vadim`
      }
    }
  ],
  adapted: [
    {
      comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      date: `2019-05-08T14:13:56.569Z`,
      id: 1,
      rating: 4,
      user: {
        avatarUrl: `img/1.png`,
        id: 4,
        isPro: false,
        name: `Max`
      }
    },
    {
      comment: `A quiet of Amsterdam.`,
      date: `2019-06-08T14:13:56.569Z`,
      id: 2,
      rating: 3,
      user: {
        avatarUrl: `img/2.png`,
        id: 5,
        isPro: false,
        name: `Vadim`
      }
    }
  ]
};
