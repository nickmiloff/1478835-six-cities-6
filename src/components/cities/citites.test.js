import React from 'react';
import {render} from '@testing-library/react';
import Test from '../test/test';

import Cities from './cities';
import {cardsFull} from '../../tests-mock';

describe(`Cities component test`, () => {
  it(`'Cities component' should render correctly with no cards`, () => {
    const CARDS = [];
    const LOCATION = `Paris`;

    const {container} = render(
        <Test>
          <Cities
            cards={CARDS}
            location={LOCATION}
            isEmpty={true} />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });

  it(`'Cities component' should render correctly with cards`, () => {
    const CARDS = cardsFull.adapted;
    const LOCATION = `Paris`;

    const {container} = render(
        <Test>
          <Cities
            cards={CARDS}
            location={LOCATION}
            isEmpty={false} />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });
});
