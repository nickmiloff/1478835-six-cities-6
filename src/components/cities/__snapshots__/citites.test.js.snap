// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities component test 'Cities component' should render correctly with cards 1`] = `
<div>
  <div
    class="cities"
  >
    <div
      class="cities__places-container container"
    >
      <section
        class="cities__places places"
      >
        <h2
          class="visually-hidden"
        >
          Places
        </h2>
        <b
          class="places__found"
        >
          3
           places to stay in 
          Paris
        </b>
        <form
          action="#"
          class="places__sorting"
          method="get"
        >
          <span
            class="places__sorting-caption"
          >
            Sort by
          </span>
          <span
            class="places__sorting-type"
            data-testid="menu"
            tabindex="0"
          >
            Popular
            <svg
              class="places__sorting-arrow"
              height="4"
              width="7"
            >
              <use
                xlink:href="#icon-arrow-select"
              />
            </svg>
          </span>
          <ul
            class="places__options places__options--custom"
          >
            <li
              class="places__option places__option--active\\""
              data-testid="option"
              tabindex="0"
            >
              Popular
            </li>
            <li
              class="places__option"
              data-testid="option"
              tabindex="0"
            >
              Price: low to high
            </li>
            <li
              class="places__option"
              data-testid="option"
              tabindex="0"
            >
              Price: high to low
            </li>
            <li
              class="places__option"
              data-testid="option"
              tabindex="0"
            >
              Top rated first
            </li>
          </ul>
        </form>
        <div
          class="cities__places-list places__list tabs__content"
        >
          <article
            class="cities__place-card place-card"
          >
            <div
              class="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              class="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/1"
              >
                <img
                  alt="Place image"
                  class="place-card__image"
                  height="200"
                  src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              class="place-card__info"
            >
              <div
                class="place-card__price-wrapper"
              >
                <div
                  class="place-card__price"
                >
                  <b
                    class="place-card__price-value"
                  >
                    €
                    161
                  </b>
                  <span
                    class="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  class="place-card__bookmark-button button"
                  data-testid="to-bookmarks-button"
                  type="button"
                >
                  <svg
                    class="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlink:href="#icon-bookmark"
                    />
                  </svg>
                  <span
                    class="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                class="place-card__rating rating"
              >
                <div
                  class="place-card__stars rating__stars"
                >
                  <span
                    style="width: 40%;"
                  />
                  <span
                    class="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                class="place-card__name"
              >
                <a
                  href="/offer/1"
                >
                  Tile House
                </a>
              </h2>
              <p
                class="place-card__type"
              >
                house
              </p>
            </div>
          </article>
          <article
            class="cities__place-card place-card"
          >
            
            <div
              class="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/2"
              >
                <img
                  alt="Place image"
                  class="place-card__image"
                  height="200"
                  src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              class="place-card__info"
            >
              <div
                class="place-card__price-wrapper"
              >
                <div
                  class="place-card__price"
                >
                  <b
                    class="place-card__price-value"
                  >
                    €
                    271
                  </b>
                  <span
                    class="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  class="place-card__bookmark-button button"
                  data-testid="to-bookmarks-button"
                  type="button"
                >
                  <svg
                    class="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlink:href="#icon-bookmark"
                    />
                  </svg>
                  <span
                    class="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                class="place-card__rating rating"
              >
                <div
                  class="place-card__stars rating__stars"
                >
                  <span
                    style="width: 72%;"
                  />
                  <span
                    class="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                class="place-card__name"
              >
                <a
                  href="/offer/2"
                >
                  Loft Studio in the Central Area
                </a>
              </h2>
              <p
                class="place-card__type"
              >
                apartment
              </p>
            </div>
          </article>
          <article
            class="cities__place-card place-card"
          >
            
            <div
              class="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/3"
              >
                <img
                  alt="Place image"
                  class="place-card__image"
                  height="200"
                  src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              class="place-card__info"
            >
              <div
                class="place-card__price-wrapper"
              >
                <div
                  class="place-card__price"
                >
                  <b
                    class="place-card__price-value"
                  >
                    €
                    865
                  </b>
                  <span
                    class="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  class="place-card__bookmark-button button place-card__bookmark-button--active"
                  data-testid="to-bookmarks-button"
                  type="button"
                >
                  <svg
                    class="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlink:href="#icon-bookmark"
                    />
                  </svg>
                  <span
                    class="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                class="place-card__rating rating"
              >
                <div
                  class="place-card__stars rating__stars"
                >
                  <span
                    style="width: 92%;"
                  />
                  <span
                    class="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                class="place-card__name"
              >
                <a
                  href="/offer/3"
                >
                  Tile House
                </a>
              </h2>
              <p
                class="place-card__type"
              >
                house
              </p>
            </div>
          </article>
        </div>
      </section>
      <div
        class="cities__right-section"
      >
        <section
          class="cities__map map"
        >
          <div
            class="leaflet-container leaflet-grab leaflet-touch-drag"
            id="map"
            style="width: 100%; height: 100%; position: relative;"
            tabindex="0"
          >
            <div
              class="leaflet-pane leaflet-map-pane"
              style="left: 0px; top: 0px;"
            >
              <div
                class="leaflet-pane leaflet-tile-pane"
              >
                <div
                  class="leaflet-layer "
                  style="z-index: 1;"
                >
                  <div
                    class="leaflet-tile-container leaflet-zoom-animated"
                    style="z-index: 18; left: 0px; top: 0px;"
                  >
                    <img
                      alt=""
                      class="leaflet-tile"
                      role="presentation"
                      src="https://a.basemaps.cartocdn.com/rastertiles/voyager/12/2074/1409.png"
                      style="width: 256px; height: 256px; left: -180px; top: -26px;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="leaflet-pane leaflet-shadow-pane"
              />
              <div
                class="leaflet-pane leaflet-overlay-pane"
              />
              <div
                class="leaflet-pane leaflet-marker-pane"
              >
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="img/pin.svg"
                  style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 34px; top: 5px; z-index: 5;"
                  tabindex="0"
                />
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="img/pin.svg"
                  style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 7436px; top: -16216px; z-index: -16216;"
                  tabindex="0"
                />
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="img/pin.svg"
                  style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 10px; top: 90px; z-index: 90;"
                  tabindex="0"
                />
              </div>
              <div
                class="leaflet-pane leaflet-tooltip-pane"
              />
              <div
                class="leaflet-pane leaflet-popup-pane"
              />
            </div>
            <div
              class="leaflet-control-container"
            >
              <div
                class="leaflet-top leaflet-left"
              />
              <div
                class="leaflet-top leaflet-right"
              />
              <div
                class="leaflet-bottom leaflet-left"
              />
              <div
                class="leaflet-bottom leaflet-right"
              >
                <div
                  class="leaflet-control-attribution leaflet-control"
                >
                  <a
                    href="https://leafletjs.com"
                    title="A JS library for interactive maps"
                  >
                    Leaflet
                  </a>
                   | © 
                  <a
                    href="https://www.openstreetmap.org/copyright"
                  >
                    OpenStreetMap
                  </a>
                   contributors © 
                  <a
                    href="https://carto.com/attributions"
                  >
                    CARTO
                  </a>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cities component test 'Cities component' should render correctly with no cards 1`] = `
<div>
  <div
    class="cities"
  >
    <div
      class="cities__places-container container cities__places-container--empty"
    >
      <section
        class="cities__no-places"
      >
        <div
          class="cities__status-wrapper tabs__content"
        >
          <b
            class="cities__status"
          >
            No places to stay available
          </b>
          <p
            class="cities__status-description"
          >
            We could not find any property available at the moment in 
            Paris
          </p>
        </div>
      </section>
      <div
        class="cities__right-section"
      />
    </div>
  </div>
</div>
`;
